kvclientpool.go

```diff
@@ -196,12 +198,37 @@ func (p *kvClientPool) startNewClientLocked() <-chan struct{} {
 
 	// create the goroutine to actually create the client
 	go func() {
-		p.logger.Debug("creating new client")
+		p.logger.Info("creating new client kv client",
+			zap.Any("clientConfig", clientConfig))
+
+		p.lock.Lock()
+		defer p.lock.Unlock()
+
+		connectErrThrottle := 1 * time.Second
+		for {
+			connectWaitPeriod := connectErrThrottle - time.Since(p.connectErrTime)
```

Comment:
Doesn't this cause a delay on initial connect?

Reply:
`connectErrTime` would be time.Time{} (0) on first connect, which yields a huge negative number and fails the next check.